packages/amplication-server/src/core/build/build.service.ts

```diff
@@ -484,7 +481,11 @@ export class BuildService {
               adminUIPath: resourceInfo.settings.adminUISettings.adminUIPath,
               serverPath: resourceInfo.settings.serverSettings.serverPath,
             },
-            pullRequestMode,
+            pullRequestMode:
+              typeof smartGitSyncEntitlement !== "boolean" &&
+              smartGitSyncEntitlement?.hasAccess
+                ? EnumPullRequestMode.Accumulative
+                : EnumPullRequestMode.Basic,
```

Comment:
I think you can do something like that:
```suggestion
          pullRequestMode: (smartGitSyncEntitlement &&
              smartGitSyncEntitlement.hasAccess &&
              EnumPullRequestMode.Accumulative) ||
            EnumPullRequestMode.Basic; 
```
or declare it above in a const and use the constant here